nsdb-core/src/main/scala/io/radicalbit/nsdb/index/AllFacetIndexes.scala

```diff
@@ -72,15 +72,52 @@ class AllFacetIndexes(basePath: String,
                         rangeFieldName: String,
                         valueFieldName: String,
                         valueFieldType: Option[IndexType[_]],
-                        ranges: Seq[TimeRange]): Seq[Bit] = facetRangeIndex.executeRangeFacet(
-    searcher,
-    query,
-    aggregationType,
-    rangeFieldName,
-    valueFieldName,
-    valueFieldType,
-    ranges
-  )
+                        ranges: Seq[TimeRange]): Seq[Bit] =
+    aggregationType match {
+      case singleAggregation: InternalTemporalSingleAggregation =>
+        facetRangeIndex.executeRangeFacet(
+          searcher,
+          query,
+          singleAggregation,
+          rangeFieldName,
+          valueFieldName,
+          valueFieldType,
+          ranges
+        )
+      case InternalAvgTemporalAggregation =>
+        val sum = facetRangeIndex.executeRangeFacet(
+          searcher,
+          query,
+          InternalSumTemporalAggregation,
+          rangeFieldName,
+          valueFieldName,
+          valueFieldType,
+          ranges
+        )
+
+        val countMap = facetRangeIndex
+          .executeRangeFacet(
+            searcher,
+            query,
+            InternalCountTemporalAggregation,
+            rangeFieldName,
+            valueFieldName,
+            valueFieldType,
+            ranges
+          )
+          .groupBy(_.timestamp)
+
+        sum.map { sumBit =>
+          countMap
+            .get(sumBit.timestamp)
+            .flatMap(_.headOption)
+            .fold(sumBit)(countBit =>
+              sumBit.copy(value = 0.0, tags = Map("count" -> countBit.value, "sum" -> sumBit.value)))
+        }
+
+      case _ =>
+        throw new RuntimeException("Not implemented yet.")
```

Comment:
we really need this case now?

Reply:
No, we actually do not need it by now. I was thinking to leave it for future aggregations, but maybe you're right and we can remove it. The match case is exhaustive, therefore the default case is not needed